maint/local_python/binding_f08.py

```diff
@@ -481,6 +489,13 @@ def process_array(p):
                 convert_list_pre.append("%s = merge(1, 0, %s)" % (arg_2, p['name']))
             if RE.match(r'out|inout', p['param_direction']):
                 convert_list_post.append("%s = (%s /= 0)" % (p['name'], arg_2))
+        elif p['_array_convert'] == "INDEX":
+            arg_1 = "%s_c" % p['name']
+            arg_2 = "%s_c" % p['name']
+            if RE.match(r'MPI_(Wait|Test)some', func['name'], re.IGNORECASE):
+                convert_list_post.append("%s(1:outcount) = %s(1:outcount) + 1" % (p['name'], arg_2))
+            else:
+                raise Exception("Unexpected in process_array: %s" % func['name'])
```

Comment:
Unexpected function?

Reply:
The `INDEX` was used with different semantics in the standard, thus I have to be careful and do them on a by-function basis. Currently, the only function with INDEX array output is `MPI_Testsome` and `MPI_Waitsome`, but in case new functions show up or I missed some function, the Exception will alert us. It is essentially an `assert`.

Reply:
I'm referring to the text. "Unexpected in process_array" is missing something, or I am not understanding what it is saying.

Reply:
When the exception occurs, I want to know which part of the code and which function. `process_array` is the name of the part. It is meant as pointers for developer debugging. Ideally, users should never see this.

Reply:
So it's an unexpected "error"? I'm just confused by the text. I get that its an exception.

Reply:
Does `Unexpected function encountered in process_array: %s` reads better?

Reply:
Yes, that sounds fine. That was my guess in the initial comment.

Reply:
Ok, will fix.

Reply:
done.